src/tape/tracked.jl

```diff
@@ -0,0 +1,213 @@
+const NULL_INDEX = typemin(Int)
+const NULL_TAPE = InstructionTape()
+
+ type TrackedReal{V<:Real,O} <: Real
```

Comment:
`immutable`? And spacing problem.

Reply:
The `value`s in the `TrackedReal` are modified as the tape is traversed, so it can't be an `immutable`